This repository is related to generation of a SAM file.
SAM is a sequence alignment file

Using guppy basecaller:

Guppy basecaller is used to create a FASTQ file for the FAST5 files of human RNA from http://s3.amazonaws.com/nanopore-human-wgs/rna/links/Bham_Run1_20171009_DirectRNA.files.txt

How to use guppy_basecaller:

The code used to perform basecalling using guppy_basecaller is

using config file: ./guppy_basecaller -i -s -c using flow cell and kit: ./guppy_basecaller -i -s --flowcell --kit

in this case I used RNA FAST5 files from Bham centre run1 of RNA with kit SQK-RNA001 and corresponding flowcell from guppy_basecaller --print_workflows FLO-MIN106 and FLO-MIN107

we can use config file rna_r9.4.1_70bps_hac based on SQK-RNA001

The output of guppy basecaller generates a FASTQ file and sequencing_summary.txt file which is used by pycoQC to generate a report.html file which contains different plots that describes the distribution of RNA.

The syntax of pycoQC is

pycoQC -f -o

Then download FASTA file of human genome from FTP ensembl. Convert gtf file of transcriptome to bed format using bedparse.

The syntax is bedparse gtf2bed file.gtf output.bed

Using bedtools getfasta, this bed file and FASTA sequence of human genome we can get FASTA sequence of RNA coding genes in human genome.

This FASTA sequence and FASTQ generated from guppy basecaller, using minimap2 these are aligned and a SAM file is generated by executing this command

minimap2 -a ref.fa test.fastq > alignment.sam
